### ISSUE TITLE
[bug] flutter_ffmpeg compatibility

### ISSUE THREAD

**Issue by jefrisingh** [Canardoux/tau#585]
## Flutter Sound Version : ^7.5.0+2
- **Released**
- **FULL** 

- Result of the command **"flutter pub deps | grep flutter_sound"**
 flutter pub deps | grep flutter_sound
|-- flutter_sound 7.5.0+2
|   |-- flutter_sound_platform_interface 7.5.0+2
|   |-- flutter_sound_web 7.5.0+2
|   |   |-- flutter_sound_platform_interface...

----------------------------------------------------------

## Severity

- Cannot build my App ? Yes

--------------------------------------------------------

## Platforms you faced the error 

- Android ? Yes

- Real device ? Yes

----------------------------------------------

## Logs

```
FAILURE: Build failed with an exception.                                
                                                                        
* What went wrong:                                                      
Execution failed for task ':app:checkReleaseDuplicateClasses'.          
> A failure occurred while executing com.android.build.gradle.internal.tasks.CheckDuplicatesRunnable
   > Duplicate class com.arthenica.mobileffmpeg.Abi found in modules jetified-mobile-ffmpeg-audio-4.3.1.LTS-runtime (com.arthenica:mobile-ffmpeg-audio:4.3.1.LTS) and jetified-mobile-ffmpeg-video-4.4-runtime (com.arthenica:mobile-ffmpeg-video:4.4)
     Duplicate class com.arthenica.mobileffmpeg.AbiDetect found in modules jetified-mobile-ffmpeg-audio-4.3.1.LTS-runtime (com.arthenica:mobile-ffmpeg-audio:4.3.1.LTS) and jetified-mobile-ffmpeg-video-4.4-runtime (com.arthenica:mobile-ffmpeg-video:4.4)
     Duplicate class com.arthenica.mobileffmpeg.BuildConfig found in modules jetified-mobile-ffmpeg-audio-4.3.1.LTS-runtime (com.arthenica:mobile-ffmpeg-audio:4.3.1.LTS) and jetified-mobile-ffmpeg-video-4.4-runtime (com.arthenica:mobile-ffmpeg-video:4.4)
     Duplicate class com.arthenica.mobileffmpeg.CameraSupport found in modules jetified-mobile-ffmpeg-audio-4.3.1.LTS-runtime (com.arthenica:mobile-ffmpeg-audio:4.3.1.LTS) and jetified-mobile-ffmpeg-video-4.4-runtime (com.arthenica:mobile-ffmpeg-video:4.4)
     Duplicate class com.arthenica.mobileffmpeg.Config found in modules jetified-mobile-ffmpeg-audio-4.3.1.LTS-runtime (com.arthenica:mobile-ffmpeg-audio:4.3.1.LTS) and jetified-mobile-ffmpeg-video-4.4-runtime (com.arthenica:mobile-ffmpeg-video:4.4)
     Duplicate class com.arthenica.mobileffmpeg.Config$1 found in modules jetified-mobile-ffmpeg-audio-4.3.1.LTS-runtime (com.arthenica:mobile-ffmpeg-audio:4.3.1.LTS) and jetified-mobile-ffmpeg-video-4.4-runtime (com.arthenica:mobile-ffmpeg-video:4.4)
     Duplicate class com.arthenica.mobileffmpeg.FFmpeg found in modules jetified-mobile-ffmpeg-audio-4.3.1.LTS-runtime (com.arthenica:mobile-ffmpeg-audio:4.3.1.LTS) and jetified-mobile-ffmpeg-video-4.4-runtime (com.arthenica:mobile-ffmpeg-video:4.4)
     Duplicate class com.arthenica.mobileffmpeg.FFprobe found in modules jetified-mobile-ffmpeg-audio-4.3.1.LTS-runtime (com.arthenica:mobile-ffmpeg-audio:4.3.1.LTS) and jetified-mobile-ffmpeg-video-4.4-runtime (com.arthenica:mobile-ffmpeg-video:4.4)
     Duplicate class com.arthenica.mobileffmpeg.Level found in modules jetified-mobile-ffmpeg-audio-4.3.1.LTS-runtime (com.arthenica:mobile-ffmpeg-audio:4.3.1.LTS) and jetified-mobile-ffmpeg-video-4.4-runtime (com.arthenica:mobile-ffmpeg-video:4.4)
     Duplicate class com.arthenica.mobileffmpeg.LogCallback found in modules jetified-mobile-ffmpeg-audio-4.3.1.LTS-runtime (com.arthenica:mobile-ffmpeg-audio:4.3.1.LTS) and jetified-mobile-ffmpeg-video-4.4-runtime (com.arthenica:mobile-ffmpeg-video:4.4)
     Duplicate class com.arthenica.mobileffmpeg.LogMessage found in modules jetified-mobile-ffmpeg-audio-4.3.1.LTS-runtime (com.arthenica:mobile-ffmpeg-audio:4.3.1.LTS) and jetified-mobile-ffmpeg-video-4.4-runtime (com.arthenica:mobile-ffmpeg-video:4.4)
     Duplicate class com.arthenica.mobileffmpeg.MediaInformation found in modules jetified-mobile-ffmpeg-audio-4.3.1.LTS-runtime (com.arthenica:mobile-ffmpeg-audio:4.3.1.LTS) and jetified-mobile-ffmpeg-video-4.4-runtime (com.arthenica:mobile-ffmpeg-video:4.4)
     Duplicate class com.arthenica.mobileffmpeg.MediaInformationParser found in modules jetified-mobile-ffmpeg-audio-4.3.1.LTS-runtime (com.arthenica:mobile-ffmpeg-audio:4.3.1.LTS) and jetified-mobile-ffmpeg-video-4.4-runtime (com.arthenica:mobile-ffmpeg-video:4.4)
     Duplicate class com.arthenica.mobileffmpeg.Packages found in modules jetified-mobile-ffmpeg-audio-4.3.1.LTS-runtime (com.arthenica:mobile-ffmpeg-audio:4.3.1.LTS) and jetified-mobile-ffmpeg-video-4.4-runtime (com.arthenica:mobile-ffmpeg-video:4.4)
     Duplicate class com.arthenica.mobileffmpeg.Statistics found in modules jetified-mobile-ffmpeg-audio-4.3.1.LTS-runtime (com.arthenica:mobile-ffmpeg-audio:4.3.1.LTS) and jetified-mobile-ffmpeg-video-4.4-runtime (com.arthenica:mobile-ffmpeg-video:4.4)
     Duplicate class com.arthenica.mobileffmpeg.StatisticsCallback found in modules jetified-mobile-ffmpeg-audio-4.3.1.LTS-runtime (com.arthenica:mobile-ffmpeg-audio:4.3.1.LTS) and jetified-mobile-ffmpeg-video-4.4-runtime (com.arthenica:mobile-ffmpeg-video:4.4)
     Duplicate class com.arthenica.mobileffmpeg.StreamInformation found in modules jetified-mobile-ffmpeg-audio-4.3.1.LTS-runtime (com.arthenica:mobile-ffmpeg-audio:4.3.1.LTS) and jetified-mobile-ffmpeg-video-4.4-runtime (com.arthenica:mobile-ffmpeg-video:4.4)
     Duplicate class com.arthenica.mobileffmpeg.util.Pair found in modules jetified-mobile-ffmpeg-audio-4.3.1.LTS-runtime (com.arthenica:mobile-ffmpeg-audio:4.3.1.LTS) and jetified-mobile-ffmpeg-video-4.4-runtime (com.arthenica:mobile-ffmpeg-video:4.4)
     Duplicate class com.arthenica.mobileffmpeg.util.Trio found in modules jetified-mobile-ffmpeg-audio-4.3.1.LTS-runtime (com.arthenica:mobile-ffmpeg-audio:4.3.1.LTS) and jetified-mobile-ffmpeg-video-4.4-runtime (com.arthenica:mobile-ffmpeg-video:4.4)
                                                                        
     Go to the documentation to learn how to <a href="d.android.com/r/tools/classpath-sync-errors">Fix dependency resolution errors</a>.
                                                                        
* Try:                                                                  
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.
                                                                        
* Get more help at https://help.gradle.org                              
                                                                        
BUILD FAILED in 43s                  
```
------------------------------------------------

**Describe the bug**
Unable to use the plugin with flutter_ffmpeg: ^0.3.0 , I hope we need to move from jetified-mobile-ffmpeg-audio-4.3.1.LTS-runtime to v. 4.4.LTS

**To Reproduce**

1. create new flutter project
2. add flutter_sound: ^7.5.0+2 
3. add flutter_ffmpeg: ^0.3.0 to pubspec.yaml
4. Try to run the app

**Expected behavior**
Build Failed

**Comment by Larpoux** [comment]
τ FULL flavor is linked to flutter_ffmpeg.
This link is in Podfile and build.gradle of the Flutter Sound plugin, so that Flutter Sound users do not have to fight with all the flutter_ffmpeg variants.

The drawback is that the App cannot add a dependency to flutter_ffmpeg in its pubspec.yaml.

Very few users are concerned by that. But I agree that the situation is not perfect.
I am not sure that there is a perfect solution ...

**Comment by jefrisingh** [comment]
Yes I agree but upgrading to latest LTS version is always a good one. Let me try to figure out the solution If I did, I'll post it here :+1:

**Comment by Larpoux** [comment]
Yeah !!!! 👍

**Comment by zartek-apps** [comment]
Same issue.. Any solution for this?

**Comment by Larpoux** [comment]
I am not happy with the actual situation neither.
I did a hard link inside the Podfile and the build.gradle because **I really wanted &tau; being able to use this fantastic library**.  But many users did not understood that and complained that they have to fight configuring their own Podfile and build.gradle.

You must understand that flutter_ffmpeg is just a wrapper above mobile_ffmpeg.
mobile_ffmpeg is very heavy and it is not possible to link it 2 times.

- If this library is linked by the App under flutter, tau-core will not be able to use it.
- If this library is linked inside tau-core, the dart App  will not be able to use it.

For example ame and libopus are embedded inside mobile-ffmpeg and tau-core will need them to encode/decode MP3 and OPUS streams **on the fly**.
I definitely want to be able to access mobile-ffmpeg from tau-core.

And ffmpeg-full-audio is just a part of the problem. Several users will want to use ffmpeg-video and will have also link-edit errors.

I will look tomorrow for a solution which will not duplicate mobile-ffmpeg, but it will not simple.

**Comment by Larpoux** [comment]
This task is correctly registered in [the roadmap](https://github.com/dooboolab/flutter_sound/projects/3).

**Comment by Larpoux** [comment]
Fixed in Flutter Sound release 7.6.3
You can [read this](https://tau.canardoux.xyz/flutter_sound_install.html#ffmpeg)
